src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/DocumentUploadServiceV2.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service;
+
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseDocumentCollection;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+
+@Service
+public class DocumentUploadServiceV2 {
```

Comment:
I don't think this name really really tells you what the purpose of the class is. Why would I use this V2 rather than the original? Is the V1 deprecated?

Reply:
Since the existing DocumentUploadService is an abstract class with a generic type parameter, it doesn't apply to my logic at all.

Reply:
Perhaps a comment at the top of v2 to explain the difference with V1?

Reply:
How about `NewUploadedDocumentsService`?

@so99y If you want to stick with the name and javadoc then ok - but the javadoc should at least make use of `@link` for class references.

Reply:
Sounds good. I will rename the service.